include/mbedtls/md.h

```diff
@@ -280,7 +280,7 @@ int mbedtls_md_starts( mbedtls_md_context_t *ctx );
  *                  Afterwards, call mbedtls_md_finish().
  *
  * \param ctx       The generic message-digest context.
- * \param input     The buffer holding the input data.
+ * \param input     The buffer holding the input data. This must not be \c NULL.
```

Comment:
This may be NULL if `ilen` is zero. 

Suggestion: This must be a readable buffer of length \p ilen Bytes. It may be \c NULL if \p ilen is zero`.